### PR TITLE
mgr/orch: pass dict to yaml.SafeDumper.represent_dict()

### PR DESCRIPTION
**Pull request by tchaikov**
yaml.SafeDumper.represent_dict() in turn calls SafeRepresenter.represent_dict().
and SafeRepresenter.represent_dict() calls BaseRepresenter.represent_mapping()
which sets `mapping` to `list(mapping.items())` before dumping it. see
https://github.com/yaml/pyyaml/blob/ee37f4653c08fc07aecff69cfd92848e6b1a540e/lib3/yaml/representer.py#L103-L129

this change alsow silences mypy errors like:

orchestrator/_interface.py: note: In member "yaml_representer" of class "DaemonDescription":
orchestrator/_interface.py:1039: error: Argument 1 to "represent_dict" of "SafeRepresenter" has incompatible type "ItemsView[Any, Any]"; expected "Mapping[Any, Any]"

Signed-off-by: Kefu Chai <kchai@redhat.com>

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Comment by tchaikov** [comment]
we have 
```
mypy run-test: commands[0] | mypy --config-file=../../mypy.ini -m alerts -m balancer -m cephadm -m crash -m dashboard -m devicehealth -m diskprediction_local -m hello -m influx -m iostat -m localpool -m mds_autoscaler -m mgr_module -m mgr_util -m mirroring -m orchestrator -m progress -m prometheus -m rbd_support -m rook -m snap_schedule -m selftest -m stats -m status -m telegraf -m telemetry -m test_orchestrator -m volumes -m zabbix
orchestrator/__init__.py:6: note: In module imported here:
orchestrator/_interface.py: note: In member "yaml_representer" of class "DaemonDescription":
orchestrator/_interface.py:1039: error: Argument 1 to "represent_dict" of "SafeRepresenter" has incompatible type "ItemsView[Any, Any]"; expected "Mapping[Any, Any]"
orchestrator/_interface.py: note: In member "yaml_representer" of class "ServiceDescription":
orchestrator/_interface.py:1178: error: Argument 1 to "represent_dict" of "SafeRepresenter" has incompatible type "ItemsView[Any, Any]"; expected "Mapping[Any, Any]"
Found 2 errors in 1 file (checked 29 source files)
ERROR: InvocationError for command /home/jenkins-build/build/workspace/ceph-pull-requests/src/pybind/mgr/.tox/mypy/bin/mypy --config-
```

after the mypy change merged recently. this change should fix it.

File: src/pybind/mgr/orchestrator/tests/test_orchestrator.py [hunk 1]
```diff
@@ -81,29 +81,29 @@ def test_apply():
...
 def test_yaml():
-    y = """daemon_type: crash
-daemon_id: ubuntu
-hostname: ubuntu
-status: 1
-status_desc: starting
-is_active: false
+    y = """daemon_id: ubuntu
+daemon_type: crash
 events:
```
Comment (sebastian-philipp) [inline src/pybind/mgr/orchestrator/tests/test_orchestrator.py:86]: Right. the order is important here for the readability.

**Comment by sebastian-philipp** [comment]
#41854

**Review by sebastian-philipp — CHANGES_REQUESTED** [review]
I'd prefer my PR. ok?

**Comment by sebastian-philipp** [comment]
closing this: It was fixed in typshed itself.

**Pull request closed without merging**